packages/core/src/tracing/instrumentation/protocols/grpcJs.js

```diff
@@ -0,0 +1,504 @@
+     * We want to use the concept of priorisation and cancel the parent span.
+     *
+     * Any other parentspan which is not node.http.server should result in an error,
+     * because it signalises that something wrong happend. Usually this case
+     * should not happen.
+     */
+    if (parentSpan) {
+      if (parentSpan.n !== 'node.http.server') {
+        logger.warn(
+          'Cannot start a GRPC-JS entry span when another span is already active. Currently, the following span is ' +
+            'active: ' +
+            JSON.stringify(parentSpan)
+        );
+        return originalHandler.apply(originalThis, originalArgs);
+      }
+
+      parentSpan.cancel();
+    }
+
+    return cls.ns.runAndReturn(() => {
+      const metadata = call.metadata;
+      const level = readMetadata(metadata, constants.traceLevelHeaderName);
+
+      if (level === '0') {
+        cls.setTracingLevel('0');
+      }
+
+      if (!isActive || cls.tracingSuppressed()) {
+        return originalHandler.apply(originalThis, originalArgs);
+      }
+
+      cls.ns.bindEmitter(call);
+
+      const incomingTraceId = readMetadata(metadata, constants.traceIdHeaderName);
+      const incomingSpanId = readMetadata(metadata, constants.spanIdHeaderName);
+
+      const span = cls.startSpan('rpc-server', constants.ENTRY, incomingTraceId, incomingSpanId);
+      span.data.rpc = {
+        call: dropLeadingSlash(name),
+        flavor: 'grpc'
+      };
+
+      if (TYPES_WITH_CALLBACK.indexOf(type) >= 0) {
+        const originalCallback = originalArgs[1];
+
+        originalArgs[1] = cls.ns.bind(function (err) {
+          if (err) {
+            span.ec = 1;
+            if (err.message || err.details) {
+              span.data.rpc.error = err.message || err.details;
+            }
+          }
+          span.d = Date.now() - span.ts;
+          span.transmit();
+          return originalCallback.apply(this, arguments);
+        });
+      }
+
+      if (TYPES_WITH_CALL_END.indexOf(type) >= 0) {
+        const originalEnd = call.end;
+
+        call.end = function () {
+          span.d = Date.now() - span.ts;
+          process.nextTick(() => {
+            // If the server emits an error, grpc calls call.end before the 'error' event handlers are processed, so we
+            // give on('error') a chance to fire and mark the span erroneous before transmitting it.
+            span.transmit();
+          });
+          return originalEnd.apply(this, arguments);
+        };
+
+        call.on('error', err => {
+          span.ec = 1;
+          if (err.message || err.details) {
+            span.data.rpc.error = err.message || err.details;
+          }
+        });
+
+        call.on('cancelled', () => {
+          span.d = Date.now() - span.ts;
+          span.transmit();
+        });
+      }
+
+      return originalHandler.apply(originalThis, originalArgs);
+    });
+  };
+}
+
+function instrumentedClientMethod(
```

Comment:
How about we move this up so it comes directly after `instrumentClient`? Then all the client side instrumentation code is together.